Microsoft.Resources/changes public preview spec

dhu5432 opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? January 2022
  3. When do you expect to publish the swagger? January 2022
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

dhu5432 committed:
New Swagger Example Spec File

dhu5432 committed:
New Swagger Spec File

dhu5432 committed:
New Readme Config File

dhu5432 committed:
Adding changes resource type:

dhu5432 committed:
PrettierCheck

dhu5432 committed:
Update settings.json

dhu5432 committed:
Swagger LintDiff

dhu5432 committed:
Merge branch 'dev-changes-Microsoft.Resources-2022-01-01' of https://github.com/dhu5432/azure-rest-api-specs into dev-changes-Microsoft.Resources-2022-01-01

dhu5432 committed:
odata.nextLink -> nextLink

dhu5432 committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

x-ms-client-flatten

# This is the commit message #2:

Moving to Microsoft.Resources namespace

Fixing ListChanges example

Reference changes.json in readme

Change Error structure and use common types resource structure

Add to properties bag

Add to properties bag

Fixing examples

Fixing examples

Prettify

Add system properties

Add system properties

Operations API implementation

Add reference to Operation

Fix operations schema

Add example

fix example

Prettify

n1b0rh00d commented on specification/changes/data-plane/Microsoft.Resources/preview/2022-01-01/changes.json:
```diff
@@ -0,0 +1,421 @@
+            "$ref": "./examples/GetChange.json"
+          }
+        },
+        "description": "Obtains the specified change resource for the target resource",
```
for the target resource change id

raosuhas commented on specification/changes/data-plane/Microsoft.Resources/preview/2022-01-01/changes.json:
```diff
@@ -0,0 +1,421 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/{resourceProviderNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Resources/changes/{changeResourceId}": {
```
>Resources [](http://example.com/codeflow?start=150&length=9)

Is this related to the Microsoft.Resources namespace from : [https://github.com/Azure/azure-rest-api-specs/tree/88e7838a09868a51de3894114355c75929847a46/specification/resources/resource-manager/Microsoft.Resources](https://github.com/Azure/azure-rest-api-specs/tree/88e7838a09868a51de3894114355c75929847a46/specification/resources/resource-manager/Microsoft.Resources)

raosuhas replied:
Please rename if this is different

raosuhas commented on specification/changes/data-plane/Microsoft.Resources/preview/2022-01-01/changes.json:
```diff
@@ -0,0 +1,421 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/{resourceProviderNamespace}/{resourceType}/{resourceName}/providers/Microsoft.Resources/changes/{changeResourceId}": {
```
>Resources [](http://example.com/codeflow?start=150&length=9)

This PR seems incomplete. Can you please send an email to armapireview@microsoft.com once it is ready ?

raosuhas reviewed (changes requested):
:clock1:

dhu5432 committed:
parent 0ebe3abc03ec2db27c00232e3340f9dbe0471048
author Daniel Hu <dhu5432@gmail.com> 1637875394 -0800
committer Daniel Hu <dhu5432@gmail.com> 1637875780 -0800

Moving to Microsoft.Resources namespace

dhu5432 committed:
Prettify

dhu5432 commented:
Moved this PR into private [repo](https://github.com/Azure/azure-rest-api-specs-pr/pull/5571) since this is for an unannounced feature
